Dynamic programming - algos/path_in_matrix.cpp

```diff
@@ -0,0 +1,87 @@
+////////////////////////////Problem statement/////////////////////////////
+
+/////
+
+Given a N X N  matrix Matrix[N][N] of positive integers.  There are only three possible moves from a cell Matrix[r][c].
+
+1. Matrix[r+1][c]
+
+2. Matrix[r+1][c-1]
+
+3. Matrix[r+1][c+1]
+
+Starting from any column in row 0, return the largest sum of any of the paths up to row N-1.
+
+Input:
+The first line of the input contains an integer T denoting the number of test cases. The description of T test cases follows.
+The first line of each test case contains a single integer N denoting the order of matrix. Next line contains N*N integers denoting the elements of the matrix in row-major form.
+
+Output:
+Output the largest sum of any of the paths starting from any cell of row 0 to any cell of row N-1. Print the output of each test case in a new line.
+
+Constraints:
+1<=T<=20
+2<=N<=20
+1<=Matrix[i][j]<=1000 (for all 1<=i<=N && 1<=j<=N)
+
+//////
+
+
+////////////////////////////Solution/////////////////////////////
+
+#include <bits/stdc++.h>
+using namespace std;
+
+
+int max_sum(vector<vector<int>> matrix)
```

Comment:
This is not a good way to pass vector to a function (here actually vector is copied). have a look at http://www.geeksforgeeks.org/passing-vector-function-cpp/  . Also between two '>>' it should be space seperated (as confuse it with bit manupulation.)